src/app.rs

```diff
@@ -1198,4 +1227,55 @@ impl App {
             }
         }
     }
+
+    pub fn get_made_for_you(&mut self) {
+        // TODO: replace searches when relevant endpoint is added
+        const DISCOVER_WEEKLY: &str = "Discover Weekly";
+        const RELEASE_RADAR: &str = "Release Radar";
+        const ON_REPEAT: &str = "On Repeat";
+        const REPEAT_REWIND: &str = "Repeat Rewind";
+
+        if self.library.made_for_you_playlists.pages.is_empty() {
+            self.made_for_you_search_and_add(DISCOVER_WEEKLY);
+            self.made_for_you_search_and_add(RELEASE_RADAR);
+            self.made_for_you_search_and_add(ON_REPEAT);
+            self.made_for_you_search_and_add(REPEAT_REWIND);
+        }
+    }
+
+    fn made_for_you_search_and_add(&mut self, search_string: &str) {
+        const SPOTIFY_ID: &str = "spotify";
+
+        if let (Some(spotify), Some(user)) = (self.spotify.clone(), self.user.clone()) {
+            let user_country =
+                Country::from_str(&user.country.unwrap_or_else(|| "".to_string())).ok();
+            match spotify.search_playlist(search_string, self.large_search_limit, 0, user_country) {
+                Ok(mut search_playlists) => {
+                    let mut filtered_playlists = search_playlists
+                        .playlists
+                        .items
+                        .iter()
+                        .filter(|playlist| {
+                            playlist.owner.id == SPOTIFY_ID && playlist.name == search_string
+                        })
+                        .map(|playlist| playlist.to_owned())
+                        .collect::<Vec<SimplifiedPlaylist>>();
+
+                    if !self.library.made_for_you_playlists.pages.is_empty() {
+                        self.library.made_for_you_playlists.pages[0]
```

Comment:
Instead of reading the array at index 0, you should use the `get_results` method implemented on the `ScrollableResultPages` struct. The will by default fetch the page in the struct's index.